pkg/tsdb/cloudmonitoring/time_series_filter.go

```diff
@@ -220,13 +223,24 @@ func (timeSeriesFilter *cloudMonitoringTimeSeriesFilter) parseResponse(queryRes
 							timeField,
 							valueField,
 						},
+						Meta: &data.FrameMeta{
+							ExecutedQueryString: executedQueryString,
+						},
 					}
 				}
 			}
 		}
 		for i := 0; i < len(buckets); i++ {
+			if buckets[i].Meta != nil {
+				buckets[i].Meta.Custom = customFrameMeta
+			} else {
+				buckets[i].SetMeta(&data.FrameMeta{Custom: customFrameMeta})
+			}
 			frames = append(frames, buckets[i])
 		}
+		if len(buckets) == 0 {
+			frames = append(frames, frame)
+		}
```

Comment:
If request is `HEADERS`, the response doesn't contain `series.Points`, buckets length becomes 0.

Reply:
Sorry, I just realized I didn't understand what you mean here.

Reply:
If Grafana set view parameter to `HEADERS` when calling `timeSeries.list`, the response doesn't contain points data.

https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.timeSeries/list#query-parameters
https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeriesView
> Returns the identity of the metric and the time series resource, but not the time series data.

In that case, the actual response becomes like following.
https://github.com/grafana/grafana/pull/45500/files#diff-5cb10f8670d719b1ecda656d9bb95821632636b8d58803a857b6a1cfa5eb5b9e

If no data points, this for loop is not executed.
https://github.com/grafana/grafana/blob/v8.4.1/pkg/tsdb/cloudmonitoring/time_series_filter.go#L164-L226

The buckets still be empty.
https://github.com/grafana/grafana/blob/v8.4.1/pkg/tsdb/cloudmonitoring/time_series_filter.go#L163

But, Query editor needs label data, Grafana should response label data.